src/main/java/org/spongepowered/api/entity/ai/task/AITask.java

```diff
@@ -75,7 +75,7 @@
      * @param other The other task
      * @return True if it can be, false if not
      */
-    boolean canRunConcurrentWith(AITask<O> other);
+    boolean canRunConcurrentWith(AITask<?> other);
```

Comment:
Sorry but it is by design that you'll always get a task passed in with the same GoalType. Change it back or provide a reason why you changed this.